src/Operations.jl

```diff
@@ -2319,7 +2319,9 @@ function test(ctx::Context, pkgs::Vector{PackageSpec};
             printpkgstyle(ctx.io, :Testing, "Running tests...")
             flush(ctx.io)
             code = gen_test_code(source_path; test_args)
-            cmd = `$(Base.julia_cmd()) $(flags) --threads=$(get_threads_spec()) --eval $code`
+            julia_args_defines_threads = any(startswith("--threads"), julia_args.exec) || (!isnothing(julia_args.env) && any(startswith("JULIA_NUM_THREADS"), julia_args.env))
+            threads_arg = julia_args_defines_threads ? `` : `--threads=$(get_threads_spec())`
+            cmd = Cmd(`$(Base.julia_cmd()) $threads_arg $(flags) --eval $code`; env = julia_args.env)
```

Comment:
This feels like a lot more change than is needed.
Don't we just need to move `--threads` to before `$flags`?

So
```
cmd = `$(Base.julia_cmd()) --threads=$(get_threads_spec()) $(flags)  --eval $code`
```
i.e.
```
% ./julia -t2 -t3 -E "Threads.nthreads(:default)"
3
```

Then the testing can be done with a normal
```
withenv(...) do
     Pkg.test(....)
end
```

Reply:
I believe I needed to also add support for passing through the environment variables to the test process in order to implement the tests...

... it could possibly be split into two PRs...

Reply:
But the last commit ("[Added @test try catch end to count Pkg.test invocations as tests](https://github.com/JuliaLang/Pkg.jl/pull/4141/commits/dfa6bbdee2980793a2a3fa1a8e4f37614258f07d)) could be reverted...

Reply:
… but yeah, the `withenv` approach might be sufficient…

Reply:
With 7aa3a807550149be325e3ac045950c9462cb4a3d, without the modification of the environment passed to the test process via `julia_args`, the tests attempting to set threads via `JULIA_NUM_THREADS` fail.

Reply:
I'd just do https://github.com/IanButterworth/Pkg.jl/commit/a2f134a48749caef7575a3c32903873775b96694

Reply:
Cherry picked :-)